### PR TITLE
GF Signup:Fix preloading conditions

### PR DESCRIPTION
**Pull request by ddc22**
Related 
- https://github.com/Automattic/growth-foundations/issues/159

## Proposed Changes
* Fix preloading conditions for the relevant plans modal experiment.
* The conditions for preloading the experiment were incorrect. All experiments were pre loaded only when there was a free subdomain. But two of these experiments actually are activated when there is a paid domain. So the code was changed to fix this conditional errors.

## Testing Instructions
* Run a few confidence checks from https://github.com/Automattic/wp-calypso/pull/80647

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/signup/steps/domains/index.jsx [hunk 1]
```diff
@@ -290,16 +290,22 @@ class DomainsStep extends Component {
 		if ( ! isPurchasingItem ) {
 			switch ( flowName ) {
 				case 'onboarding':
-					loadExperimentAssignment(
-						'calypso_gf_signup_onboarding_free_free_dont_miss_out_modal_v3'
-					);
-					loadExperimentAssignment( 'calypso_onboarding_plans_paid_domain_on_free_plan' );
+					if ( isPurchasingItem ) {
```
Comment (southp) [inline client/signup/steps/domains/index.jsx]: The condition above is checking for `! isPurchasingItem`, so there is no chance this condition works :) 

Also, could you explain the intention behind this change? That way we can better test and review the code.

Comment (ddc22) [reply client/signup/steps/domains/index.jsx]: Thanks for raising this James! I have fixed it!

Comment (ddc22) [reply client/signup/steps/domains/index.jsx]: > Also, could you explain the intention behind this change? That way we can better test and review the code.

Actually the conditions for preloading the experiment were incorrect. All experiments were pre loaded only when there was a free subdomain. But two of these experiments actually are activated when there is a paid domain.

**Review by aaronyan — APPROVED** [review]
LGTM! I've tested both flows and was able to assign myself to `treatment` and get the modal for a paid domain + free plan.

**Pull request merged**